Updating auth dev docs (#2944)
chef/automate: dev-docs/auth.md

Our work on Authentication and Authorization is based on industry best-practices.

- [OpenID Connect](http://openid.net/connect/)
- [AWS IAM Policies](http://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies.html)

We evaluated a set of tools that would meet our requirements in [Authentication and Authorization in Automate Spike](https://github.com/chef/aaa-spike) and decided on:

- [Dex](https://github.com/coreos/dex) as OpenID solution

> We added our own [authn-service](../components/authn-service) as glue between those services to add support for non-human clients. If you're interessted, please read the [design documentation](../components/authn-service/DESIGN.md).

Reviewer: ```suggestion
We added our own [authn-service](../components/authn-service) as glue between those services to add support for non-human clients. If you're interested, please read the [design documentation](../components/authn-service/DESIGN.md).
```
🤷‍♀ while we're here

  (This doesn't mean that we don't pick that route in the future.)

* There's an example Go method in [`docs/examples/api_request.go`](examples/api_request.go).

## Using the Identity and Access Management System in the Automate UI

> By default any new user, team, or non-admin api token is denied access to everything. They all need to be added to polciies in order to be granted permissions. 

Reviewer 2: ```suggestion
By default any new user, team, or non-admin API token is denied access to everything. They all need to be added to policies in order to be granted permissions. 
```

> By default any new user, team, or non-admin API token is denied access to everything. They all need to be added to policies in order to be granted permissions. 

Author: ```suggestion
By default, any new user, team, or non-admin API token is denied access to everything. They all need to be added to teams or policies in order to be granted permission. 
```

### Adding a New User to a New Policy
1. Create an admin token using 
    ```

>     export TOKEN=`chef-automate iam token create <your-token-name-here> --admin`

Reviewer 3: [sanity check] - Not sure if this is just my case, but I wasn't able to run this as a one liner.  

Instead I ran `chef-automate iam token create <your-token-name-here> --admin` and then copied the result and ran `export TOKEN=<result>`

Author: ohhh thats so weird, whats the output it gives you when you try and run it as one line?

Reviewer 3: `bash: export: --admin: not a valid identifier`

Reviewer 3: ahh...wait ok..  It works if I use `export TOKEN='chef-automate iam token create sometoken --admin'`

the difference being using apostrophes

Author: so the backticks done work? hmm

Reviewer 3: scratch that sorry - apostrophes did not work either

Author: i think apostrophes would just save that whole command into token, right?

Reviewer 3: and here I am now and backticks are working.

I think i'm done with Reviews for the day.  Well done!

Author: lol okay cool, i wonder what was going on before. glad its working now!

Reviewer 3: I just triple checked using backticks, exactly correct. 🙏 

I think we can put what I was getting back as 100% user error 😂

    ```

> 1. Create a new role, this role will allow access to the Event Feed, Applications, and Projects list.

Reviewer 2: ```suggestion
1. Create a new role; this role will allow access to the Event Feed, Applications, and Projects list.
```

    ```

>     curl -k -sSH "api-token: $TOKEN" -d '{"name": "Example Role","id": "example-role", "actions": ["event:*", "applications:*", "iam:projects:list"]}' -X POST https://a2-dev.test/apis/iam/v2beta/roles?pretty

Reviewer 2: ```suggestion
    curl -k -sSH "api-token: $TOKEN" -d '{"name": "Example Role","id": "example-role", "actions": ["event:*", "applications:*", "iam:projects:list"]}' -X POST https://a2-dev.test/apis/iam/v2/roles?pretty
```

    ```
1. Create a new policy using the role you just created. You can find out more about [creating policies](https://automate.chef.io/docs/iam-v2-guide/#creating-custom-policies) in our documentation.
    ```

>     curl -k -sSH "api-token: $TOKEN" -d '{"name": "Example Policy", "id": "example-policy", "statements": [{"effect": "ALLOW", "role": "example-role", "projects": ["*"]}]}' -X POST https://a2-dev.test/apis/iam/v2beta/policies?pretty

Reviewer 2: ```suggestion
    curl -k -sSH "api-token: $TOKEN" -d '{"name": "Example Policy", "id": "example-policy", "statements": [{"effect": "ALLOW", "role": "example-role", "projects": ["*"]}]}' -X POST https://a2-dev.test/apis/iam/v2/policies?pretty
```

    ```
1. Sign in with the username `admin` and password `chefautomate`
1. Navigate to Settings > Users and create a new user
1. Navigate to Settings > Policies > the policy you just created > Members
1. Add the user you just created as a member of the policy

> 1. Sign out as admin and in as your new user

Reviewer 2: ```suggestion
1. Sign out as admin and sign in as your new user
```

> 1. You should see the Event Feed, Applications, and Settings top-nav items, as that is what the user you just created has access to.

Reviewer 3: Typo - whole list came out as "1."

> 1. You should see the Event Feed, Applications, and Settings top-nav items, as that is what the user you just created has access to.

Reviewer 3: Typo - this whole section came out as "1."

Author: thats standard markdown syntax so that you can add items later without having to renumber the same list, or are you calling out something else that im overlooking?

Reviewer 3: I stand corrected.  All is good here

Reviewer 2: Yes, it is legal but I have to voice my vote that it is slightly less readable for those who want to view the file "raw". To me it is not much burden for having to renumber manually since if one is in there editing you are mentally doing that already. (Just my opinion; not a big deal.)
